src/client/renderingengine.cpp

```diff
@@ -187,243 +173,44 @@ void RenderingEngine::cleanupMeshCache()
-	// gtk/gdk code (gdk/x11/gdksurface-x11.c) for the setup_top_level
-	// method. But for now (as it would require some significant changes)
-	// leave the code as is.
-
-	// The following is borrowed from the above gdk source for setting top
-	// level windows. The source indicates and the Xlib docs suggest that
-	// this will set the WM_CLIENT_MACHINE and WM_LOCAL_NAME. This will not
-	// set the WM_CLIENT_MACHINE to a Fully Qualified Domain Name (FQDN) which is
-	// required by the Extended Window Manager Hints (EWMH) spec when setting
-	// the _NET_WM_PID (see further down) but running Minetest in an env
-	// where the window manager is on another machine from Minetest (therefore
-	// making the PID useless) is not expected to be a problem. Further
-	// more, using gtk/gdk as the model it would seem that not using a FQDN is
-	// not an issue for modern Xorg window managers.
-
-	verbosestream << "Client: Setting Xorg window manager Properties"
-		<< std::endl;
-
-	XSetWMProperties (x11_dpl, x11_win, NULL, NULL, NULL, 0, NULL, NULL, NULL);
-
-	// Set the _NET_WM_PID window property according to the EWMH spec. _NET_WM_PID
-	// (in conjunction with WM_CLIENT_MACHINE) can be used by window managers to
-	// force a shutdown of an application if it doesn't respond to the destroy
-	// window message.
-
-	verbosestream << "Client: Setting Xorg _NET_WM_PID extended window manager property"
-		<< std::endl;
-
-	Atom NET_WM_PID = XInternAtom(x11_dpl, "_NET_WM_PID", false);
-
-	pid_t pid = getpid();
-
-	XChangeProperty(x11_dpl, x11_win, NET_WM_PID,
-			XA_CARDINAL, 32, PropModeReplace,
-			reinterpret_cast<unsigned char *>(&pid),1);
-
-	// Set the WM_CLIENT_LEADER window property here. Minetest has only one
-	// window and that window will always be the leader.
-
-	verbosestream << "Client: Setting Xorg WM_CLIENT_LEADER property"
-		<< std::endl;
-
-	Atom WM_CLIENT_LEADER = XInternAtom(x11_dpl, "WM_CLIENT_LEADER", false);
-
-	XChangeProperty (x11_dpl, x11_win, WM_CLIENT_LEADER,
-		XA_WINDOW, 32, PropModeReplace,
-		reinterpret_cast<unsigned char *>(&x11_win), 1);
-#endif
-}
-
-#ifdef _WIN32
-static bool getWindowHandle(irr::video::IVideoDriver *driver, HWND &hWnd)
-{
-	const video::SExposedVideoData exposedData = driver->getExposedVideoData();
-
-	switch (driver->getDriverType()) {
-	case video::EDT_OGLES1:
-	case video::EDT_OGLES2:
-	case video::EDT_OPENGL:
-		hWnd = reinterpret_cast<HWND>(exposedData.OpenGLWin32.HWnd);
-		break;
-	default:
-		return false;
-	}
-
-	return true;
-}
-#endif
-
 bool RenderingEngine::setWindowIcon()
 {
-#if defined(XORG_USED)
+	irr_ptr<video::IImage> img;
+	auto try_img_path = [&](const std::string &path) {
+		if (!img)
+			img.reset(driver->createImageFromFile(path.c_str()));
+	};
+
 #if RUN_IN_PLACE
-	return setXorgWindowIconFromPath(
-			porting::path_share + "/misc/" PROJECT_NAME "-xorg-icon-128.png");
+	try_img_path(porting::path_share + "/misc/" PROJECT_NAME "-xorg-icon-128.png");
```

Comment:
this file is not installed on windows so this doesn't work currently

would make sense to move it to `textures/base/pack` imo

Reply:
We already have a `logo.png` file there. It's 100x100 pixels, which is maybe not a good size. But using it seems to work for me.